### PR TITLE
Customize handling 404 not found status

### LINKED ISSUE #181: Provide a way to customize handling response statuses
Frozen Flask has `FREEZER_IGNORE_404_NOT_FOUND`, which warns on 404 responses but saves the response given, a bit like the `save` policy for redirects.

A similar mechanism as for redirects (3xx) should preferably be used for Not Found (404, 4xx). It should be possible to use a custom function, or a predefined action.

### PR DESCRIPTION
**Pull request by nappex**
fix #181

### CODE REVIEW

**Comment by nappex** [comment]
- [ ] update readme
- [ ] make tests
- [ ] make the code

File: tests/fixtures/app_404_not_found/app.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from flask import Flask
+
+app = Flask(__name__)
+freeze_config = {'ignore_404_not_found': True}
```
Comment (encukou) [inline tests/fixtures/app_404_not_found/app.py:4]: I'd rather use something like:
```suggestion
freeze_config = {
    'status_handlers': {
        '201': 'warn',
        '301': 'follow',
        '404': 'ignore',
        '418': 'my_module:custom_handler',
        '429': 'retry',
        '5xx': 'error',
    }
}
```

**Comment by nappex** [comment]
1. Consider to move new func to outside of class. Easier to test by unit test, maybe you could try `import freezer.status_handler` as method for tests. Realize if you move func you has to add config parameter to handle configuration of statuses.
2. Add unit tests for `status_handler`
3. Create a way how to parametrize integration test with one app to handle a few statuses and policies in one configuration.
4. Tame the wild beat  with name `mypy`

**Comment by nappex** [comment]
to discuss - https://www.w3docs.com/snippets/git/how-to-force-git-pull-to-override-local-files.html

**Comment by nappex** [comment]
- handler funcs should get low level objects as arguments ?
- problem with interface func `follow` other func do not need arguments, it should be consistent ?
- problem with return value - should be raise Error or string as value of status see func `save`

File: README.md [hunk 2]
```diff
@@ -315,24 +315,56 @@ The object has the following attributes:
...
+* `'warn'`: warn message to stdout and skip
 * `'save'`: `freezeyt` will save the body of the redirect page, as if
   the response was `200`.
-  This is meant to be used with redirects in the HTML `<meta>` tag.
-* `'follow'`: `freezeyt` will save content from the redirected location.
-* `'ignore'`: `freezeyt` will not save any content for pages that redirect.
+* `'follow'`: `freezeyt` will save content from the redirected location (handler to redirect statuses 3XX).
+* `'ignore'`: `freezeyt` will not save any content for pages
+* `'error'` (default): When certain status response is encountered,
+  `freezeyt` will abort.
```
Comment (encukou) [inline README.md:328]:
```suggestion
* `'error'`: abort with an error
```

**Comment by nappex** [comment]
I am not sure about handler `save`, it just return status...

File: freezeyt/status_handlers.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    print(
+        f"WARNING: URL {task.get_a_url()}, status code: {status[:3]} was freezed"
+    )
+
+
+def follow(status: str, task: TaskInfo) -> None:
+    try:
+        location = task._task.response_headers['Location']
+    except KeyError:
+        raise UnexpectedStatus(task.get_a_url(), status, task._task.reasons)
```
Comment (encukou) [inline freezeyt/status_handlers.py:16]:
```suggestion
    location = task._task.response_headers['Location']
```

File: freezeyt/util.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ def __init__(self, url, status, reasons=None):
         self.url = url
```
Comment (encukou) [inline freezeyt/util.py:26]:
```suggestion
        self.url = str(url)
```

File: tests/fixtures/app_status_handler/app.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
+from flask import Flask, Response
+
+from freezeyt.freezer import IgnorePage
+from freezeyt.util import parse_absolute_url
+from freezeyt.hooks import TaskInfo
+
+
+def custom_status_handler(status: str, task: TaskInfo) -> None:
+    task._freezer.add_task(parse_absolute_url('http://localhost:8000/404.html'))
```
Comment (encukou) [inline tests/fixtures/app_status_handler/app.py:9]:
```suggestion
    task.freeze_info.add_url('http://localhost:8000/404.html')
```

File: tests/fixtures/app_status_handler/app.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
+from flask import Flask, Response
+
+from freezeyt.freezer import IgnorePage
+from freezeyt.util import parse_absolute_url
```
Comment (encukou) [inline tests/fixtures/app_status_handler/app.py:4]:
```suggestion
```

File: tests/fixtures/circular_redirect/app.py [hunk 7]
```diff
@@ -1,7 +1,9 @@
 from flask import Flask, redirect, url_for
 
 app = Flask(__name__)
-freeze_config = {}
+freeze_config = {
+    'status_handlers': {'302': 'error'}
+}
```
Comment (encukou) [inline tests/fixtures/circular_redirect/app.py:6]:
```suggestion
freeze_config = {}
```

**Review by encukou — APPROVED** [review]

**Pull request merged**